src/controllers/controller.doughnut.js

```diff
@@ -138,6 +138,37 @@ module.exports = DatasetController.extend({
 		return ringIndex;
 	},
 
+	// Get radius length offset of the dataset in relation to the visible datasets weights. This allows determining the inner and outer radius correctly
+	getRingWeightedOffset: function(datasetIndex) {
+		var ringWeightOffset = 0;
+
+		for (var j = 0; j < datasetIndex; ++j) {
+			if (this.chart.isDatasetVisible(j)) {
+				ringWeightOffset += this.getRingWeight(j);
+			}
+		}
+
+		return ringWeightOffset;
+	},
+
+	getRingWeight: function(dataSetIndex) {
+		var ringWeight = this.chart.data.datasets[dataSetIndex].weight;
+		if (!ringWeight) {
+			ringWeight = 1;
+		}
+		return ringWeight;
+	},
+
+	getVisibleDatasetTotalWeight: function() {
+		var totalWeight = 0;
```

Comment:
I think you could replace this entire function with:

    return getRingWeightedOffset(this.chart.data.datasets.length);

Reply:
Thanks! I should have realized that one myself.